lib/ret_web/controllers/api-internal/v1/login_email_controller.ex

```diff
@@ -0,0 +1,23 @@
+defmodule RetWeb.ApiInternal.V1.LoginEmailController do
+  use RetWeb, :controller
+
+  def update(conn, %{"old_email" => old_email, "new_email" => new_email}) do
+    case Ret.change_email_for_login(%{old_email: old_email, new_email: new_email}) do
+      :ok ->
+        json(conn, %{success: true})
+
+      {:error, reason} ->
+        conn
+        |> put_status(status_code_for_error()[reason] || :internal_server_error)
+        |> json(%{error: reason})
+    end
+  end
+
+  defp status_code_for_error() do
+    [
+      new_email_already_in_use: :conflict,
+      no_account_for_old_email: :not_found,
+      invalid_parameters: :bad_request
+    ]
+  end
```

Comment:
Your original was idiomatic; it’s just that the `Status.code/1` call wasn’t necessary. You can just do this:
```elixir
defp status_code(:new_email_already_in_use),
  do: :conflict

defp status_code(:no_account_for_old_email),
  do: :not_found

defp status_code(:invalid_parameters),
  do: :bad_request

defp status_code(_error_reason),
  do: :internal_server_error

```